tests/test_mailchimp_start_date.py

```diff
@@ -0,0 +1,188 @@
+                primary_keys_sync_2 = set(primary_keys_list_2)
+
+                if self.expected_metadata()[stream][self.OBEYS_START_DATE]:
+
+                    # Collect information specific to incremental streams from syncs 1 & 2
+                    expected_replication_key = next(
+                        iter(self.expected_replication_keys().get(stream, []))
+                    )
+                    replication_dates_1 = [
+                        row.get("data").get(expected_replication_key)
+                        for row in synced_records_1.get(stream, {"messages": []}).get(
+                            "messages", []
+                        )
+                        if row.get("data")
+                    ]
+                    replication_dates_2 = [
+                        row.get("data").get(expected_replication_key)
+                        for row in synced_records_2.get(stream, {"messages": []}).get(
+                            "messages", []
+                        )
+                        if row.get("data")
+                    ]
+
+                    # Verify replication key is greater or equal to start_date for sync 1
+                    for replication_date in replication_dates_1:
+
+                        self.assertGreaterEqual(
+                            self.parse_date(
+                                replication_date, self.RECORD_DATETIME_FORMAT
+                            ),
+                            self.parse_date(self.start_date_1, self.START_DATE_FORMAT),
+                            msg="Report pertains to a date prior to our start date.\n"
+                            + "Sync start_date: {}\n".format(self.start_date_1)
+                            + "Record date: {} ".format(replication_date),
+                        )
+
+                    # Verify replication key is greater or equal to start_date for sync 2
+                    for replication_date in replication_dates_2:
+
+                        self.assertGreaterEqual(
+                            self.parse_date(
+                                replication_date, self.RECORD_DATETIME_FORMAT
+                            ),
+                            self.parse_date(self.start_date_2, self.START_DATE_FORMAT),
+                            msg="Report pertains to a date prior to our start date.\n"
+                            + "Sync start_date: {}\n".format(self.start_date_2)
+                            + "Record date: {} ".format(replication_date),
+                        )
+                    # Verify the number of records replicated in sync 1 is greater than the number
+                    # of records replicated in sync 2
+                    self.assertGreater(record_count_sync_1, record_count_sync_2)
+
+                    # Verify the records replicated in sync 2 were also replicated in sync 1
+                    self.assertTrue(primary_keys_sync_2.issubset(primary_keys_sync_1))
```

Comment:
Why we are not asserting on equality of primary keys? If that is expected please add the reason comment.

Reply:
```
                    # Verify by primary key the same records are replicated in the 1st and 2nd syncs
                    self.assertSetEqual(primary_keys_sync_1, primary_keys_sync_2)
```
Line No: 188